lib/ansible/modules/windows/win_disk_management.ps1

```diff
@@ -0,0 +1,1411 @@
+                }
+                Check {
+                    $CheckService = (Get-Service ShellHWDetection).Status -eq "Running"
+
+                    return $CheckService
+                }
+                default {
+                    throw "Neither '-Stop', '-Start' nor 'Check' switch was passed to the function when invoked, without the switches the service can not be maintained"
+                }
+        }
+}
+
+function Create-Partition {
+        param(
+                $Disk,
+                $SetDriveLetter
+        )
+
+        $Partition = $Disk | New-Partition -UseMaximumSize -DriveLetter $SetDriveLetter
+
+        return $Partition
+}
+
+function Create-Volume {
+        param(
+                $Volume,
+                $FileSystem,
+                $FileSystemLabel,
+                $FileSystemAllocUnitSize,
+                $FileSystemLargeFRS,
+                $FileSystemShortNames,
+                $FileSystemIntegrityStreams
+        )
+
+        $Alloc = $FileSystemAllocUnitSize *1KB
+
+        $ParaVol = @{
+            FileSystem = $FileSystem
+            NewFileSystemLabel = $FileSystemLabel
+            AllocationUnitSize = $Alloc
+        }
+
+        if ($FileSystem -eq "ntfs") {
+            if (-not $FileSystemLargeFRS) {
+                $CreatedVolume = $Volume | Format-Volume @ParaVol -ShortFileNameSupport $FileSystemShortNames -Force -Confirm:$false
+
+                return $CreatedVolume
+            } else {
+                $CreatedVolume = $Volume | Format-Volume @ParaVol -UseLargeFRS -ShortFileNameSupport $FileSystemShortNames -Force -Confirm:$false
+
+                return $CreatedVolume
+            }
+        } elseif ($FileSystem -eq "refs") {
+            $CreatedVolume = $Volume | Format-Volume @ParaVol -SetIntegrityStreams $FileSystemIntegrityStreams -Force -Confirm:$false
+
+            return $CreatedVolume
+        }
+}
+
+# Rescan disks
+try {
+    $null = Invoke-Expression '"rescan" | diskpart'
+} catch {
+    $result.general_log.rescan_disks = "failed"
+}
+$result.general_log.rescan_disks = "successful"
+
+$ParamsDisk = @{
+    DiskSizeInteger = $SizeInteger
+    DiskNumberInteger = $NumberInteger
+    DiskSize = "$(if ($SizeInteger) {
+                                $SizeF = $Size *1GB
+                            } else {
+                                $SizeF = $null
+                            }
+                           )$SizeF"
+    PartitionStyle = $FindPartitionStyle
+    OperationalStatus = $OperationalStatus
+    ReadOnly = $ReadOnly
+    Number = $Number
+}
+
+# Search disk
+try {
+    $disk = Search-Disk @ParamsDisk
+} catch {
+    $result.general_log.search_disk = "failed"
+    Fail-Json -obj $result -message "Failed to search and/or select the disk with the specified option values: $($_.Exception.Message)"
+}
+if ($disk) {
+    $diskcount = $disk | Measure-Object | Select-Object  -ExpandProperty Count
+    if ($diskcount -ge 2) {
+        $disk = $disk[0]
```

Comment:
Should you not be failing here instead of just picking the first disk?

Reply:
no, because this will only happen if no number option was passed to the module and exactly the same disks were found with the select options. Then it doesn't matter which disk will be chosen and I select the first found.